src/pipecat/processors/aggregators/llm_response_universal.py

```diff
@@ -443,101 +375,52 @@ async def _handle_transcription(self, frame: TranscriptionFrame):
-                    )
-                    self._emulating_vad = False
-            finally:
-                self._aggregation_event.clear()
-
-    async def _maybe_emulate_user_speaking(self):
-        """Maybe emulate user speaking based on transcription.
-
-        Emulate user speaking if we got a transcription but it was not
-        detected by VAD. Behavior when bot is speaking depends on the
-        enable_emulated_vad_interruptions parameter.
-        """
-        # Check if we received a transcription but VAD was not able to detect
-        # voice (e.g. when you whisper a short utterance). In that case, we need
-        # to emulate VAD (i.e. user start/stopped speaking), but we do it only
-        # if the bot is not speaking. If the bot is speaking and we really have
-        # a short utterance we don't really want to interrupt the bot.
-        if (
-            not self._user_speaking
-            and not self._waiting_for_aggregation
-            and len(self._aggregation) > 0
-        ):
-            if self._bot_speaking and not self._params.enable_emulated_vad_interruptions:
-                # If emulated VAD interruptions are disabled and bot is speaking, ignore
-                logger.debug("Ignoring user speaking emulation, bot is speaking.")
-                await self.reset()
-            else:
-                # Either bot is not speaking, or emulated VAD interruptions are enabled
-                # - trigger user speaking emulation.
-                await self.push_frame(EmulateUserStartedSpeakingFrame(), FrameDirection.UPSTREAM)
-                self._emulating_vad = True
+
+    async def _on_user_turn_started(self, strategy: BaseUserTurnStartStrategy):
+        await self._trigger_user_turn_start(strategy)
+
+    async def _on_bot_turn_started(self, strategy: BaseBotTurnStartStrategy):
+        await self._trigger_bot_turn_start(strategy)
+
+    async def _on_push_frame(
+        self,
+        strategy: BaseUserTurnStartStrategy | BaseBotTurnStartStrategy,
+        frame: Frame,
+        direction: FrameDirection = FrameDirection.DOWNSTREAM,
+    ):
+        await self.push_frame(frame, direction)
+
+    async def _on_broadcast_frame(
+        self,
+        strategy: BaseUserTurnStartStrategy | BaseBotTurnStartStrategy,
+        frame_cls: Type[Frame],
+        **kwargs,
+    ):
+        await self.broadcast_frame(frame_cls, **kwargs)
+
+    async def _trigger_user_turn_start(self, strategy: BaseUserTurnStartStrategy):
+        # Reset all user turn start strategies to start fresh.
+        if self.turn_start_strategies:
+            for s in self.turn_start_strategies.user:
+                await s.reset()
+
+        if self._params.enable_user_speaking_frames:
+            logger.debug(f"User started speaking (user turn start strategy: {strategy})")
+            await self.broadcast_frame(UserStartedSpeakingFrame, emulated=strategy is None)
```

Comment:
Not sure I follow the `emulated=strategy is None` logic...it should be considered emulated if it's not triggered by the VAD strategy, right?

Reply:
Yes, this should come from the event instead of here. That is, the user start turn transcription strategy should set this to true.

Reply:
Actually, I think emulated should be completely removed. I don't think there's a use in knowing this anymore.

Reply:
Done.

Reply:
So...this is a (bit) of a breaking change, is it not? Now if someone is using a pipeline in its default configuration and cared about reading the `emulated` value out of `User*SpeakingFrame`s, they'd...what? Only ever see the default `false` value?

Reply:
> I don't think there's a use in knowing this anymore.

Out of curiosity, why was it useful before?

I get that now that there can be arbitrarily many strategies to trigger the user turn start, a boolean is unsuitable for describing which one actually triggered it.

If this information _is_ still useful on the frame for some reason, though, we _could_ consider supplanting the boolean field for one that contains an identifier for the strategy.